templates/templates/component/cdktf/tsconfig.json.tmpl

```diff
@@ -1,39 +1,37 @@
 {
   "compilerOptions": {
-    "composite": true,
     "baseUrl": ".",
-    "rootDir": "src",
     "outDir": "dist",
-    "alwaysStrict": true,
+    "declaration": true,
+    "declarationMap": true,
     "experimentalDecorators": true,
     "sourceMap": true,
     "inlineSources": true,
-    "lib": ["es2018"],
-    "target": "es6",
-    "module": "NodeNext",
-    "moduleResolution": "NodeNext",
-    "noFallthroughCasesInSwitch": true,
-    "noImplicitAny": true,
-    "noImplicitReturns": true,
-    "noImplicitThis": true,
+    "lib": ["es2023"],
+    "target": "es2022",
+    "module": "esnext",
+    "moduleResolution": "bundler",
```

Comment:
module should be `node16` ?

[ref](https://github.com/tsconfig/bases/blob/main/bases/node20.json)